### PR TITLE
ASM Mesh id should be proj-${PROJECTNUBMER}

### PR DESCRIPTION
**Pull request by jlewi**
* Fix kubeflow/gcp-blueprints#117
* The pattern for the mesh id needs to be proj-${PROJECTNUMBER}
* Dedube the mesh substitutions; there were two different substitutions "mesh-id" and "asm-mesh-id" for the mesh id.

### CODE REVIEW

File: gcp/v2/asm/istio-operator.yaml [hunk 1]
```diff
@@ -25,14 +25,14 @@ spec:
       istio-ingressgateway:
         type: NodePort
     global:
-      meshID: "project-id_us-central1_name" # {"$kpt-set":"asm-mesh-id"}
+      meshID: "proj-147474701642" # {"$kpt-set":"asm-mesh-id"}
```
Comment (Bobgy) [inline gcp/v2/asm/istio-operator.yaml:28]: You probably forgot to reset the id?

Comment (jlewi) [reply gcp/v2/asm/istio-operator.yaml]: Good catch. I updated the gcp-reset make command and reran it. A bunch of other values got reset as well.

File: gcp/Kptfile [hunk 2]
```diff
@@ -20,17 +20,17 @@ openAPI:
...
         setter:
           name: gcloud.compute.region
-          value: us-central1
+          value: REGION
     io.k8s.cli.setters.location:
       x-k8s-cli:
         setter:
           name: location
-          value: us-east1-d
+          value: location
```
Comment (Bobgy) [inline gcp/Kptfile:33]: nit: shall we make the default value's casing consistent?

**Review by Bobgy — COMMENTED** [review]
Thank you!
/LGTM
/Approve

One nitpicking, but I think we can first get this in.

**Pull request merged**